app/src/main/java/com/ayush/weatherapp/main/MainActivity.java

```diff
@@ -77,4 +86,8 @@ private void setNavigationView() {
           }
         });
   }
+
+  @Override public void setText(String s) {
```

Comment:
setText() is very generic. The methods should be particular like setTemperature or set/showError.

Reply:
:ok: